src/Illuminate/Support/Collection.php

```diff
@@ -726,6 +726,24 @@ public function forPage($page, $perPage)
     }
 
     /**
+     * Output a collection with two elements. Items in the first element did pass
+     * the given $callback, items in the second element did not.
+     *
+     * @param  callable $callback
+     * @return static
+     */
+    public function partition(callable $callback)
+    {
+        $partitions = [new static(), new static()];
+
+        foreach ($this->items as $item) {
+            $partitions[! (int) $callback($item)][] = $item;
+        }
+
+        return new static($partitions);
```

Comment:
Shouldn't you just return the plain array so you can separate them using `lists`?